syntax = "proto3";
package Payment;

message Order
{
	message ProductInfo
	{
		string productId = 1;
		string productName = 2;
		int32 amount = 3;
		double price = 4;
	}
	string serviceHost = 1;
	string storeId = 2;
	string choosePayment = 3;
	string token1 = 4;
	string token2 = 5;
	string orderId = 6;
	string userId = 7;
	string orderTime = 8;
	string returnURL = 9;
	string remoteIPAddress = 10;
	string groupId = 20;
	string repeatType = 21;
	string totalRepeatCount = 22;
	repeated ProductInfo orderList = 50;
	string desc = 51;

	message Result
	{
		string code = 1;
		string url = 2;
		string uid = 3;
		string key = 4;
		string form = 5;
		string action = 6;
	}

	message Error
	{
		string errCode = 1;
		string message = 2;
	}
}

message CheckOrder
{
	string serviceHost = 1;
	string storeId = 2;
	string token1 = 3;
	string token2 = 4;
	string uid = 5;
	string key = 6;

	message Result
	{
		string code = 1;
		string message = 2;
		string uid = 3;
		string key = 4;
		string orderId = 10;
		string userId = 11;
		string pfn = 12;
		string finishtime = 13;
		string cost = 15;
		string paymentName = 20;

		repeated string cancel_order = 50;
	}

	message Error
	{
		string errCode = 1;
		string message = 2;
		string uid = 3;
		string key = 4;
		string orderId = 10;
		string userId = 11;
		string pfn = 12;
		string finishtime = 13;
		string paymentName = 14;
		string cost = 15;
	}
}

message CancelOrder
{
	string serviceHost = 1;
	string storeId = 2;
	string token1 = 3;
	string token2 = 4;
	string orderId = 5;
	string groupId = 6;
	string cancelTime = 7;
	string cancelReason = 8;

	message Result
	{
		string code = 1;
		string message = 2;
		string orderId = 3;
		string groupId = 4;
	}

	message Error
	{
		string errCode = 1;
		string message = 2;
	}
}

message RealtimePaymentReport
{
	string key 					=1;
	string prc 					=2;
	string cardno 				=3;
	string acode 				=4;
	string order_id 			=5;
	string user_id 				=6;
	string uid 					=7;
	string cost 				=8;
	string currency 			=9;
	string actual_cost 			=10;
	string actual_currency 		=11;
	string love_cost 			=12;
	string retmsg 				=13;
	string pfn 					=14;
	string finishtime 			=15;
	string payment_name 		=16;
	string nois 				=17;
	string group_id 			=18;
	string echo_0 				=19;
	string echo_1 				=20;
	string echo_2 				=21;
	string echo_3 				=22;
	string echo_4 				=23;
	string redeem				=24;
	string trans_type			=25;
}

message NonRealtimePaymentReport
{
	string key 					=1;
	string prc 					=2;
	string acode 				=3;
	string order_id 			=4;
	string user_id 				=5;
	string uid 					=6;
	string cost 				=7;
	string currency 			=8;
	string actual_cost 			=9;
	string actual_currency 		=10;
	string love_cost 			=11;
	string retmsg 				=12;
	string pfn 					=13;
	string finishtime 			=14;
	string payment_name 		=15;
	string nois 				=16;
	string group_id 			=17;
	string echo_0 				=18;
	string echo_1 				=19;
	string echo_2 				=20;
	string echo_3 				=21;
	string echo_4 				=22;
}

message OrderConfirmReport
{
	string key 					=1;
	string prc 					=2;
	string order_id 			=3;
	string user_id 				=4;
	string uid 					=5;
	string cost 				=6;
	string currency 			=7;
	string actual_cost 			=8;
	string actual_currency 		=9;
	string love_cost 			=10;
	string retmsg 				=11;
	string pfn 					=12;
	string finishtime 			=13;
	string payment_name 		=14;
	string nois 				=15;
	string group_id 			=16;
	string echo_0 				=17;
	string echo_1 				=18;
	string echo_2 				=19;
	string echo_3 				=20;
	string echo_4 				=21;
}